lib/testutils/backends.go

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2016-2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutils
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"os/exec"
+
+	etcdclient "github.com/coreos/etcd/client"
+	consulapi "github.com/hashicorp/consul/api"
+
+	. "github.com/onsi/ginkgo/extensions/table"
+
+	"github.com/projectcalico/libcalico-go/lib/api"
+	"github.com/projectcalico/libcalico-go/lib/client"
+	"os"
+)
+
+// CleanBackend is a utility function to wipe clean "/calico" recursively from backend.
+func CleanBackend(config *api.CalicoAPIConfig) {
+	var err error
+
+	log.Println(fmt.Sprintf("Cleaning datastore: %v", config.Spec.DatastoreType))
+
+	switch config.Spec.DatastoreType {
+	case api.EtcdV2:
+		cfg := etcdclient.Config{
+			Endpoints: []string{config.Spec.EtcdScheme + "://" + config.Spec.EtcdAuthority},
+		}
+		c, _ := etcdclient.New(cfg)
+		kapi := etcdclient.NewKeysAPI(c)
+		_, err = kapi.Delete(context.Background(), "/calico", &etcdclient.DeleteOptions{Dir: true, Recursive: true})
+	case api.ConsulV1:
+		cfg := consulapi.Config{
+			Address: config.Spec.ConsulAddress,
+			Scheme:  config.Spec.ConsulScheme,
+		}
+		c, _ := consulapi.NewClient(&cfg)
+		kv := c.KV()
+		ops := consulapi.KVTxnOps{
+			&consulapi.KVTxnOp{Key: "calico", Verb: consulapi.KVDelete},
+			&consulapi.KVTxnOp{Key: "calico/", Verb: consulapi.KVDeleteTree},
+		}
+		_, _, _, err = kv.Txn(ops, nil)
+	case api.Kubernetes:
```

Comment:
I think we should either error here just so this doesn't get used by k8s tests without fixing it up.